dali/pipeline/data/dltensor.cc

```diff
@@ -12,13 +12,102 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cuda_runtime_api.h>
+#include <condition_variable>
+#include <list>
+#include <mutex>
 #include <string>
+#include <thread>
 #include "dali/pipeline/data/dltensor.h"
 #include "dali/core/error_handling.h"
+#include "dali/core/mm/detail/aux_alloc.h"
 #include "dali/core/static_switch.h"
 
 namespace dali {
 
+class DLTensorGraveyard {
```

Comment:
Maybe add some docs why we need this and how it works?

Reply:
Will do.

Reply:
Done.